tests/unit/swagger/schemaDetails.spec.ts

```diff
@@ -183,6 +183,49 @@ describe('Schema details generation', () => {
 
     describe('methods', () => {
       describe('responses', () => {
+        it('should generate headers from method reponse decorator.', () => {
+          const metadata = new MetadataGenerator('./fixtures/controllers/responseHeaderController.ts').Generate();
+          const responseSpec = new SpecGenerator2(metadata, getDefaultExtendedOptions()).GetSpec();
+          const paths = ['SuccessResponseWithHeaderClass', 'SuccessResponseWithObject', 'ResponseWithHeaderClass', 'ResponseWithObject', 'TsoaResponseWithHeaderClass', 'TsoaResponseWithObject'];
+          paths.forEach((path: string) => {
+            const responses = responseSpec.paths[`/ResponseHeader/${path}`].get?.responses;
+
+            expect(responses?.[200]?.headers).to.not.eq(undefined);
+            if (path.includes('HeaderClass')) {
+              expect(responses?.[200]?.headers).to.deep.eq({
+                Link: {
+                  type: 'string',
+                  description: 'a link string',
```

Comment:
Should have `required: true`.

Reply:
There is no `required` field in [HeaderObject at OpenAPI2.0](https://swagger.io/specification/v2#header-object)

Reply:
You're right, this was supposed to a comment related to the OAS3 specs, which are now properly asserting this.